apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx

```diff
@@ -109,7 +105,7 @@ const useScrub = (options: {
     });
 
     return scrub.disconnectedCallback;
-  }, [options.value.type]);
+  }, [options.value.type, options.value?.unit]);
```

Comment:
I would use `options.value` inside the `useEffect` body instead of using `optionsRef.current.value`, so that eslint would detect it as a dependency and catch bugs like this.

Maybe would also put only `onChange` and `onChangeComplete` into a ref to make more clear the intention of ignoring changes of the callbacks identity.

Doesn't really affect behaviour, so not super important if hard to do for some reason, just would be a bit more correct.

Reply:
The problem is that we are calling onChange here and that onChange is going to update the value, so we end up in an infinite loop, I agree though what I did doesn't look nice and clear at all

Reply:
trying something out rn

Reply:
ok I tried to make it better, not sure I succeeded